### ISSUE TITLE
Duplicate traits are created when altering the payment method

### ISSUE THREAD

**Issue by sdgun** [topcoder-platform/member-api#54]
Duplicate traits are created when altering the payment method more than once for a particular user within polling duration (3 days). We noticed dynamodb has only one record, but somehow ES is receiving multiple records

Example users: taasintake1000, testsdgun

Example response:
[
    {
        "traitId": "onboarding_checklist",
        "categoryName": "Onboarding Checklist",
        "traits": {
            "traitId": "onboarding_checklist",
            "data": [
                {
                    "user_payment_method": {
                        "date": 1633497003327,
                        "message": "success",
                        "payment_method": "wipro payroll",
                        "status": "completed"
                    }
                }
            ]
        }
    },
    {
        "traitId": "onboarding_checklist",
        "categoryName": "Onboarding Checklist",
        "traits": {
            "traitId": "onboarding_checklist",
            "data": [
                {
                    "user_payment_method": {
                        "status": "completed",
                        "message": "success",
                        "date": 1633497003327,
                        "payment_method": "wipro payroll"
                    }
                }
            ]
        }
    }
]

**Comment by Schpotsky** [comment]
@sdgun ,

Cannot re-produce the issue, please see this video: https://monosnap.com/direct/SUVbF5eXiktXWgi3BR0jA9cPaDg2mp

For testing, I setup the following apps locally (on another machine):

- tc-bus-api
- member-api
- member-processor-es

Can you provide more details about how to re-produce the issue ? specifically the initial traits for the users used for testing

cc @veshu

**Comment by sdgun** [comment]
@Schpotsky Attached the initial status of the given test users before we changed the payment method
<img width="446" alt="taasintake1000 initial status" src="https://user-images.githubusercontent.com/4592712/136368709-390935dd-3527-4ab0-a6f7-d5ad983fe09c.png">

<img width="425" alt="testsdgun initial status" src="https://user-images.githubusercontent.com/4592712/136368714-bfa7f67f-8dab-44e8-8922-a37cd1f8d351.png">

The step we followed was generating the payment method change event for these users twice during a period of like 30 minutes.

**Comment by Schpotsky** [comment]
> @Schpotsky can you please try to make two changes for the same user? I mean our job should receive one record for user in one run and then two records in the next run.

Getting two records for the same user will not happen, we only get the one with latest modify_date [GET_MODIFIED_PAYMENT_METHODS_QUERY](https://github.com/topcoder-platform/onboarding-processor/blob/develop/src/services/PaymentMethodsProcessorService.js#L17 )

Can you provide an extract of the user_payment_method table in the dev environment for the above users ? 
```sql
select * from user_payment_method where user_id in (88778061,40152238);
```

Also the corresponding data in Elasticsearch for the profile traits for these users will be helpful:
index: membertraits-2020-01
type: profiletraits
```
GET {{ElasticSearch-Host}}:9200/membertraits-2020-01/profiletraits/_search?q=userId:88778061
GET {{ElasticSearch-Host}}:9200/membertraits-2020-01/profiletraits/_search?q=userId:40152238
```

@vikasrohit

**Comment by Schpotsky** [comment]
@vikasrohit ,

It is an issue related to the member-api itself, it is not related to onboarding-processor.

Duplicate traits are created when calling `PUT {{URL}}/members/{{handle}}/traits`, I re-produced the issue in member-api by deploying the `feature/shapeup4-cqrs-standards-update` branch locally.

It looks like this is the branch which is deployed on the dev environment, can you double check ?

Also let me know, whether I should fix it in the above mentioned branch or this will be done by the team working on the CQRS feature.

Thanks

**Comment by vikasrohit** [comment]
Yes you are right @Schpotsky, that branch is deployed into dev right now. Let me ask @sachin-maheshwari how he want to fix the issue with that branch for duplicate traits.
